### PR TITLE
Handle not found case for status update

### PR DESCRIPTION
**Pull request by cpuguy83**
If we don't handle the "pod not found" case then we end up with the pod
getting re-queued over and over until the max retries are hit. It also
blocks the queue for other pod status updates for that pod
namespace/name.

### CODE REVIEW

**Comment by cpuguy83** [comment]
I wonder if the pod status queue should be using the pod UID instead of namespace/name.

File: vkubelet/pod.go [hunk 1]
```diff
@@ -204,7 +205,7 @@ func (s *Server) updatePodStatus(ctx context.Context, pod *corev1.Pod) error {
 	ctx = addPodAttributes(ctx, span, pod)
 
 	status, err := s.provider.GetPodStatus(ctx, pod.Namespace, pod.Name)
-	if err != nil {
+	if err != nil && !strongerrors.IsNotFound(err) {
```
Comment (sargun) [inline vkubelet/pod.go]: Can you add to the docs somewhere that `GetPod` / `GetPodStatus` should return a `NotFoundError`?

Comment (cpuguy83) [reply vkubelet/pod.go]: See #635

**Comment by sargun** [comment]
How can you guarantee pod names are unique? Can't you delete a pod, and recreate it with the same name?

**Comment by pires** [comment]
Pod names are unique. You can't have two pods with the same name in the API. Now, it may be possible for multiple different instances of the VK to have each their instance of a pod with the same name, but that only means the _illegal_ VK instance(s) are handling stale data.

**Comment by sargun** [comment]
@pires Kind of. If you have a network partition, a pod can be deleted and recreated with the same name. Although this may seem weird, it can happen if people are using pod names a kind of "singleton" key.

**Comment by sargun** [comment]
Again, it's okay for VK1.0 to use namespace/name.

File: vkubelet/queue.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func handleQueueItem(ctx context.Context, q workqueue.RateLimitingInterface, han
 		if err := handler(ctx, key); err != nil {
 			if q.NumRequeues(key) < maxRetries {
 				// Put the item back on the work queue to handle any transient errors.
-				log.G(ctx).Warnf("requeuing %q due to failed sync: %v", key, err)
+				log.G(ctx).WithError(err).Warnf("requeuing %q due to failed sync", key)
```
Comment (jeremyrickard) [inline vkubelet/queue.go]: 👍

**Comment by pires** [comment]
I don't think Kubernetes has core primitives that prevent a scenario where certain kubelets (or any other controllers) stop seeing an(y) API server and continue to run workloads that the kube-scheduler will event reschedule to other nodes (visible from an API perspective).
Don't forget API servers are stateless. They don't care about network partitions. For as long as an API server has access to an healthy etcd, it has visibility into _everything_ that is stored there, including (un)healthy nodes.

**Pull request merged**